persistence-api/src/main/java/org/apache/cassandra/stargate/schema/TableMetadata.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.cassandra.stargate.schema;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Table information.
+ *
+ * <p>Backend agnostic representation equivalent to {@code
+ * org.apache.cassandra.schema.TableMetadata}
+ */
+public interface TableMetadata {
+  /** Gets the table id. */
+  UUID getId();
+
+  /** Gets the case-sensitive name of the keyspace. */
+  String getKeyspace();
+
+  /** Gets the case-sensitive name of the table. */
+  String getName();
+
+  List<ColumnMetadata> getPartitionKeys();
+
+  List<ColumnMetadata> getClusteringKeys();
+
+  /** A collection of static and regular columns. */
+  List<ColumnMetadata> getColumns();
+
+  /**
+   * Gets an object that is used as identity for this instance. This object will be used to
+   * represent this version of the table metadata for caching.
+   */
+  Object getIdentity();
```

Comment:
Isn't it possible to create a class that represents this entity? (with the table name, schema name)

Reply:
I think this can be resolved once we know what the persistence api would look like.

The idea here is to provide a caching mechanism for `TableMetadata`. In C*, `TableMetadata` is immutable, so it should be straightforward, but if we wrap it each time in a new instance for the backend agnostic API, we loose that immutability. If we use the wrapping approach, returning the private instance on `getIdentity()` would solve that.